crates/vm/tracing.rs

```diff
@@ -0,0 +1,116 @@
+use bytes::Bytes;
+use ethrex_common::{serde_utils, H256};
+use ethrex_common::{types::Block, Address, U256};
+use serde::Serialize;
+
+use crate::{backends::revm::REVM, Evm, EvmError};
+
+/// Collection of traces of each call frame as defined in geth's `callTracer` output
+/// https://geth.ethereum.org/docs/developers/evm-tracing/built-in-tracers#call-tracer
+pub type CallTrace = Vec<Call>;
+
+/// Trace of each call frame as defined in geth's `callTracer` output
+/// https://geth.ethereum.org/docs/developers/evm-tracing/built-in-tracers#call-tracer
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Call {
+    /// Type of the Call
+    pub r#type: CallType,
+    /// Address that initiated the call
+    pub from: Address,
+    /// Address that received the call
+    pub to: Address,
+    /// Amount transfered
+    pub value: U256,
+    /// Gas provided for the call
+    #[serde(with = "serde_utils::u64::hex_str")]
+    pub gas: u64,
+    /// Gas used by the call
+    #[serde(with = "serde_utils::u64::hex_str")]
+    pub gas_used: u64,
+    /// Call data
+    #[serde(with = "serde_utils::bytes")]
+    pub input: Bytes,
+    /// Return data
+    #[serde(with = "serde_utils::bytes")]
+    pub output: Bytes,
+    /// Error returned if the call failed
+    pub error: Option<String>,
+    /// Revert reason if the call reverted
+    pub revert_reason: Option<String>,
+    /// List of nested sub-calls
+    pub calls: Vec<Call>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    /// Logs (if enabled)
+    pub logs: Option<Vec<CallLog>>,
+}
+
+#[derive(Serialize, Debug)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
```

Comment:
```suggestion
#[serde(rename_all = "UPPERCASE")]
```
Just a nit, I've seen [geth](https://geth.ethereum.org/docs/developers/evm-tracing/built-in-tracers) using uppercase for calls and it looks a little bit better IMO 😄